src/ray/raylet/lib/python/common_extension.cc

```diff
@@ -283,19 +297,22 @@ static PyMethodDef PyObjectID_methods[] = {
      "Return the object ID as a string in hex."},
     {"is_nil", (PyCFunction)PyObjectID_is_nil, METH_NOARGS,
      "Return whether the ObjectID is nil"},
-    {"__reduce__", (PyCFunction)PyObjectID___reduce__, METH_NOARGS,
-     "Say how to pickle this ObjectID. This raises an exception to prevent"
-     "object IDs from being serialized."},
+    {"__reduce__", (PyCFunction)PyObjectID___reduce__, METH_VARARGS,
```

Comment:
So, my concern was that people would define remote functions that captured object IDs and that most of the time this happened it would be an accident.

I'm not really sure how much this kind of error would occur, since I haven't seen too many people complaining about object IDs not being pickleable on GitHub.

It does force us to do some ugly stuff to make actor handles pickleable (since actor handles include a bunch of object IDs).

I could go either way on this one. @guoyuhong what were your reasons for making them pickleable? Is it to simplify the actor handle code?

Reply:
I see there is https://github.com/ray-project/ray/issues/1317.